frontend/src/composables/formAutoSave.ts

```diff
@@ -0,0 +1,46 @@
+import { ref, onMounted, onBeforeUnmount } from 'vue';
+
+// autosave functionality for forms, saves on inactivity after the delay (default 10 seconds)
+export function useAutoSave(saveFunc: () => Promise<void>, delay: number = 10000) {
```

Comment:
Optional Style: Not sure on naming schemes, but technically the `saveFunc` argument behaves as a wrapped callback function. Could consider naming it `saveCallback` or even just `callback` as it could align with more conventional JS library naming schemes.